### ISSUE TITLE
Create a p5.js file for the cheatsheet instead of indesign!

### ISSUE THREAD

**Issue by bmoren** [bmoren/p5js-cheat-sheet#12]

**Comment by bmoren** [comment]
this should be HTML for printability & accessibility

**Comment by limzykenneth** [comment]
Follow up from the [forum](https://discourse.processing.org/t/a-p5-js-cheat-sheet-for-beginners/8236/21), you can have a look at the source code of what I did [here](https://github.com/limzykenneth/p5-cheat-sheet-html).

Let me know how you would like to merge that work and I'll be happy to file a PR as necessary.

**Comment by bmoren** [comment]
Hi Kenneth, sorry for the delay on this. 
This looks fantastic! Let's unify this. into one place. If you want to make a PR here, I'm happy to make this the default. I do think that we should keep pdf versions (these can just be 'print to pdf' from the web version) for folks who are trying to access offline. One change I'd like to see is for the desktop version of the HTML site to better reflect the design of the original sheet rather than a mobile view. but if you want to merge here, I can work on that. 

Thanks for your work on this!

**Comment by bmoren** [comment]
I think we can deploy to github pages and I'm happy to buy and support a domain name as well.

**Comment by limzykenneth** [comment]
I'll merge the `nl` translation first then I'll prepare a PR here to merge that in. Do let me know how you would like the repo structure to be though since you will need to handle the deployment to github pages as well.

> I do think that we should keep pdf versions (these can just be 'print to pdf' from the web version) for folks who are trying to access offline.

Do you mean pre-prepared pdfs? Right now if you save from the print dialog (or print to a pdf printer) it should be formatted to fit exactly on a piece of paper.

> One change I'd like to see is for the desktop version of the HTML site to better reflect the design of the original sheet rather than a mobile view.

The HTML structure is there so it shouldn't be too hard. Just ping me when you are working on this if you need any help.

**Comment by vincentsijben** [comment]
Hi guys,

I've added dutch translation into the repo from kenneth. One commit was about Ben's usage of x1 and y1 in ellipse, rect and arc. The usage of that "1" implies there is also a x2. Imho it should be just `rect(x,y,width,height);` instead of `rect(x1,y1,width,height);`. What do you think?

**Comment by bmoren** [comment]
I do mean per-prepared PDFs, I know folks can print them, but I do think there is value for folks who are looking to get offline (w/o printing) asap in having those as well (and to keep the links that were out there alive!). As for the HTML, it shouldn't be hard to adjust the online layout, you've got a good initial structure here.

@vincentsijben it makes sense to me that it'd be changed to x/y in rect, arc, and ellipse. I'll get that corrected on everything once things migrate here!

For the repo structure I'm pretty open, but on first glance (and depending on our preferences) I think it would make sense for the root to be the pre-compiled pdf files and keep the HTML files inside of `/docs` and then github pages can be hooked into that folder?

Another option would be to keep the HTML files in the root and have the PDF files in a `/pdf` folder to preference the HTML version which is maybe better, but a bit less nice to look at from the github view. 

another thought: did you have any ideas for the domain name?!
p5jscheatsheet.com / .org / .dev 
p5js.codes
p5js.tools
p5js.guide
p5js.tips
p5js.dev

**Comment by bmoren** [comment]
as for what's currently in the repo, I think we can basically wipe it out and replace it with the work you've done, or migrate all of the old stuff into a archive branch.

**Comment by vincentsijben** [comment]
I guess @limzykenneth can send a PR to you? And you can organize the repo the way you want. I'd be happy to help once merged.

Maybe wait with the domainname. You already rank first with your github repo. It's promoting yourself as well :)

**Comment by limzykenneth** [comment]
>  I think it would make sense for the root to be the pre-compiled pdf files and keep the HTML files inside of /docs and then github pages can be hooked into that folder?

I think that made sense both from a github point of view and html point of view, looking at a github repo with just html files isn't immediately helpful, let's go with that.

> as for what's currently in the repo, I think we can basically wipe it out and replace it with the work you've done, or migrate all of the old stuff into a archive branch.

An archive branch would be nice just for posterity sake.

For domain name p5jscheatsheet.com sounds ideal to me. The p5js.* domains feel like they can be confusing when compared to p5js.org so I would avoid those.

I'll see if I can prepare the PR tomorrow.

**Comment by vincentsijben** [comment]
> @vincentsijben it makes sense to me that it'd be changed to x/y in rect, arc, and ellipse. I'll get that corrected on everything once things migrate here!

Maybe you could also consider all the built-in functions to end with ";". I can see my students literally copying those statements, forgetting the ;. Just nice coding, in the processing equivalent a must-have.
